lib/puppet/pops/types/types.rb

```diff
@@ -2799,6 +2800,12 @@ def really_instance?(o, guard = nil)
     resolved_type.really_instance?(o, guard)
   end
 
+  # @return `nil` to prevent serialization of the type_expr used when first initializing this instance
+  # @api private
+  def type_expr
+    nil
```

Comment:
Why is this needed for serialiation - later types are backed by Pcore and I assume pcore is directing the serialization.
Mostly curious.